pkg/discovery/kubernetes/control-point-discovery.go

```diff
@@ -0,0 +1,178 @@
+package kubernetes
+
+import (
+	"context"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/fluxninja/aperture/pkg/etcd/election"
+	"github.com/fluxninja/aperture/pkg/k8s"
+	"github.com/fluxninja/aperture/pkg/log"
+	"github.com/fluxninja/aperture/pkg/panichandler"
+	"golang.org/x/exp/slices"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/cache"
+)
+
+func newControlPointDiscovery(election *election.Election, k8sClient k8s.K8sClient, controlPointStore ControlPointStore) (*controlPointDiscovery, error) {
+	cpd := &controlPointDiscovery{
+		election:          election,
+		controlPointStore: controlPointStore,
+		discoveryClient:   k8sClient.GetClientSet().DiscoveryClient,
+		dynamicClient:     k8sClient.GetDynamicClient(),
+	}
+
+	return cpd, nil
+}
+
+// controlPointDiscovery is a struct that helps with Kubernetes control point discovery.
+type controlPointDiscovery struct {
+	waitGroup         sync.WaitGroup
+	ctx               context.Context
+	cancel            context.CancelFunc
+	controlPointStore ControlPointStore
+	discoveryClient   discovery.DiscoveryInterface
+	dynamicClient     dynamic.Interface
+	election          *election.Election
+}
+
+// Start starts the Kubernetes control point discovery.
+func (cpd *controlPointDiscovery) start() {
+	cpd.ctx, cpd.cancel = context.WithCancel(context.Background())
+
+	cpd.waitGroup.Add(1)
```

Comment:
The waitgroup is not being used ? 
 wait() is not being called

Reply:
It's being waited on in [stop()](https://github.com/fluxninja/aperture/blob/f0ea05d0ced70abc420e17835c1afac3bf32ebe7/pkg/discovery/kubernetes/control-point-discovery.go#L177) to make sure the goroutine ends gracefully at shutdown